lib/internal/main/check_syntax.js

```diff
@@ -76,7 +73,7 @@ function checkSyntax(source, filename) {
   // Remove BOM.
   source = stripBOM(source);
   // Wrap it.
-  source = CJSModule.wrap(source);
```

Comment:
Should we actually continue to use the wrapper here? If I am not mistaken, it should be possible to use `compileFunction` directly here (as we do for for all CJS modules in case the wrapper is not manipulated):

`const { compileFunction } = internalBinding('contextify');`.

Reply:
I don't think we could do that because there is a test that specifically test for monkey-patching the CJS loader in `-c` with `-r`...then this will be sort-of semver-major.

Reply:
We could check if it's patched or not identical to the other code but that's probably best done in another PR in that case.